### PR TITLE
refactor(internal-plugin-metrics): move rtcMetrics from plugin-meetings

### PR DESCRIPTION
**Pull request by rarajes2**
# COMPLETES #< [SPARK-529191](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-529191) - part-1 >

## This pull request addresses

Moves the `rtcMetrics` from `plugin-meetings` to `internal-plugin-metrics`.

## by making the following changes

`RtcMetrics` could be used from `internal-plugin-metrics` to any other packages needed. It's no longer specific just to `plugin-meetings`.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios where tested

After moving to internal-plugin-metrics, tested the meeting on samples page. It's working as expected.
Please find the attached HAR file on the JIRA. Check this [comment](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-529191?focusedId=6537507&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-6537507) for reference.

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/test/unit/spec/roap/request.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import 'jsdom-global/register';
```
Comment (Kesari3008) [inline packages/@webex/plugin-meetings/test/unit/spec/roap/request.ts:1]: Is this change required as part of this PR?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts [hunk 2]
```diff
@@ -749,11 +749,12 @@ describe('internal-plugin-metrics', () => {
           `name: client.alert.displayed`,
         ]);
 
-        assert.deepEqual(webexLoggerLogCalls[2].args, [
-          'call-diagnostic-events -> ',
-          'CallDiagnosticMetrics: @createClientEventObjectInMeeting => collected browser data',
-          '{"error":"unable to access window.navigator.userAgent"}',
-        ]);
+        // Invesitgate why this is failing
```
Comment (Kesari3008) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:752]: Are we planning to take care of this in separate PR ?

**Comment by Kesari3008** [comment]
Please update the test section in the PR decsription

File: packages/@webex/plugin-meetings/test/unit/spec/roap/request.ts (hunk 1 above)
Comment (rarajes2) [reply packages/@webex/plugin-meetings/test/unit/spec/roap/request.ts:1]: Yes, this was required as we were getting `window is not defined` error. Discussed with Anna as well, this is the fix.

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts (hunk 2 above)
Comment (rarajes2) [reply packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:752]: Yes, we should investigate this separately. I will create a [SPARK-566674](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-566674) to look into this.

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 3]
```diff
@@ -2449,9 +2449,11 @@ describe('plugin-meetings', () => {
           checkWorking({allowMediaInLobby: true});
         });
 
-        it('should create rtcMetrics and pass them to Media.createMediaConnection()', async () => {
+        // Stubbing a named import class is not working. Need to investigate further.
+        it.skip('should create rtcMetrics and pass them to Media.createMediaConnection()', async () => {
```
Comment (rarajes2) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:2453]: Created a JIRA to investigate this failing UT.

**Review by Kesari3008 — APPROVED** [review]

**Review by chrisadubois — COMMENTED** [review]
code looks fine but i feel like we should understand why unit tests which were passing before are now failing before merging this... thats kinda the point of unit tests

**Comment by rarajes2** [comment]
> code looks fine but i feel like we should understand why unit tests which were passing before are now failing before merging this... thats kinda the point of unit tests

@chrisadubois - One UT is failing because of the way export is changed. Another is failing because we are getting the `window.navigator.userAgent` which was earlier not available in the UT, reason for this could be - We have added `jsdom-global/register`. We can look into this separately as this is little tricky to fix, I have created a JIRA for it. Anna is already investigating `jsdom-global` issue.

**Comment by chrisadubois** [comment]
right... but now we are talking about merging without the confidence these UTs provide - to what end? just to be done with this? why don't we resolve those issues for the UTs or replace them with an equal value test instead of just saving it for later? We need that confidence, no?

File: packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts [hunk 4]
```diff
@@ -752,7 +755,9 @@ describe('internal-plugin-metrics', () => {
         assert.deepEqual(webexLoggerLogCalls[2].args, [
           'call-diagnostic-events -> ',
           'CallDiagnosticMetrics: @createClientEventObjectInMeeting => collected browser data',
-          '{"error":"unable to access window.navigator.userAgent"}',
+          `{"_ua":${JSON.stringify(
+            userAgent
+          )},"parsedResult":{"browser":{"name":"webex-js-sdk","version":"test-webex-version client=Cantina;"},"os":{},"platform":{},"engine":{}}}`,
```
Comment (rarajes2) [inline packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.ts:760]: This change is because of `jsdom-global` which adds the window object and now we can get the `userAgent`.

**Comment by rarajes2** [comment]
> right... but now we are talking about merging without the confidence these UTs provide - to what end? just to be done with this? why don't we resolve those issues for the UTs or replace them with an equal value test instead of just saving it for later? We need that confidence, no?

@chrisadubois  - Fixed both the UTs.

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 5]
```diff
@@ -2472,18 +2468,14 @@ describe('plugin-meetings', () => {
...
           assert.calledOnce(Media.createMediaConnection);
           assert.calledWith(
             Media.createMediaConnection,
             true,
             meeting.getMediaConnectionDebugId(),
             meeting.id,
-            sinon.match({
-              rtcMetrics: fakeRtcMetrics,
-            })
+            sinon.match.hasNested('rtcMetrics.webex', webex)
```
Comment (rarajes2) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:2478]: Here we can't mock the constructor because of the named import so we don't get the property `default`. So checking if the `rtcMetrics` is present and has property `webex` with same mock instance.

**Review by chrisadubois — APPROVED** [review]

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js (hunk 3 above)
Comment (rarajes2) [reply packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js]: Closed this JIRA as we have fixed these UT as part of this PR itself.
